src/Microsoft.AspNet.Mvc.Core/Rendering/Html/HtmlHelper.cs

```diff
@@ -50,6 +53,7 @@ public class HtmlHelper : IHtmlHelper, ICanHasViewContext
             HtmlEncoder = htmlEncoder;
             UrlEncoder = urlEncoder;
             JavaScriptStringEncoder = javaScriptStringEncoder;
+            _logger = loggerFactory.CreateLogger<HtmlHelper>();
```

Comment:
`loggerFactory.CreateLogger(GetType());` if you want to support the derived type scenarios we spoke about.